### PR TITLE
67 userprofile page

### PR DESCRIPTION
**Pull request by oleksandr-vovk**
user-profile

Temporary uses user with id 1.

### CODE REVIEW

**Review by ismv — CHANGES_REQUESTED** [review]
Conflicts Resolving

File: backend/Tasque.BLL/Services/FileUploadService.cs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        {
+            return fileData.Split(';')[0].Split(':')[1];
+        }
+
+        private string GetExtention(string contentType)
+        {
+            return contentType.Split('/')[1];
+        }
+
+        private string CreateFileName(string extettion)
```
Comment (ismv) [inline backend/Tasque.BLL/Services/FileUploadService.cs]: extension

Comment (oleksandr-vovk) [reply backend/Tasque.BLL/Services/FileUploadService.cs]: fixed

File: backend/Tasque.BLL/Options/AmazonS3Options.cs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+﻿
+namespace Tasque.Core.BLL.Options
+{
+    public class AmazonS3Options
+    {
+        public string AwsAccessKey { get; set; } = null!;
+        public string AwsSecretAccessKey { get; set; } = null!;
+        public string BucketName { get; set; } = null!;
+        public string Region { get; set; } = null!;
+    }
```
Comment (Anatolii-Ivanov) [inline backend/Tasque.BLL/Options/AmazonS3Options.cs:10]: Make sure you and Danil will have the same config, as we will use one AWS account for everything

File: backend/Tasque.BLL/Services/FileUploadService.cs [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        private readonly AmazonS3Options _options;
+        public FileUploadService(IOptions<AmazonS3Options> options)
+        {
+            _options = options.Value;
+            _bucketName = _options.BucketName;
+            _awsS3Client = new AmazonS3Client(_options.AwsAccessKey, _options.AwsSecretAccessKey, RegionEndpoint.GetBySystemName(_options.Region));
+        }
+
+		public async Task<string?> UploadFileAsync(string fileData, string folder)
+		{
```
Comment (Anatolii-Ivanov) [inline backend/Tasque.BLL/Services/FileUploadService.cs]: Formatting

File: backend/Tasque.BLL/Services/UserService.cs [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+            var userEntity = await _context.Users.FirstOrDefaultAsync(u => u.Id == dto.Id)
+                ?? throw new ValidationException("User not found");
+            userEntity.Name = dto.Name;
+            if(string.IsNullOrEmpty(dto.AvatarURL) || dto.AvatarURL.StartsWith("http"))
+            {
+                userEntity.AvatarURL = dto.AvatarURL;
+            }
+            else
+            {
+                userEntity.AvatarURL = await _fileUploadService.UploadFileAsync(dto.AvatarURL, "avatars");
```
Comment (Anatolii-Ivanov) [inline backend/Tasque.BLL/Services/UserService.cs:53]: Can you name avatar (as I understand in base64 format)  with more meaningful name? Not AvatarURL

You can create UserEditDto for such purpose

File: backend/Tasque.DAL/Migrations/20220817073025_AddedUserAvatar.cs [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+        }
+
+        protected override void Down(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.DropColumn(
+                name: "AvatarURL",
+                table: "Users");
+        }
+    }
+}
```
Comment (Anatolii-Ivanov) [inline backend/Tasque.DAL/Migrations/20220817073025_AddedUserAvatar.cs:25]: Update Readme scheme with your changes

File: frontend/src/app/user/components/user-profile/user-profile.component.ts [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+      userNameControl: this.userNameControl,
+      emailControl: this.emailControl
+    });
+    this.changePassForm = new FormGroup({
+      prevPasswordControl: this.prevPasswordControl,
+      newPasswordControl: this.newPasswordControl
+    });
+  }
+
+  public handleFileInput(target: any): void {
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/user/components/user-profile/user-profile.component.ts]: github actions

**Review by Anatolii-Ivanov — CHANGES_REQUESTED** [review]

File: backend/Tasque.BLL/Services/UserService.cs (hunk 4 above)
Comment (oleksandr-vovk) [reply backend/Tasque.BLL/Services/UserService.cs:53]: I use the same string field for url and file data in base64 format. If it has URL or null, the value srores to db, and if it has file the data saves to S3 and generated link saves to db.

File: backend/Tasque.BLL/Services/FileUploadService.cs (hunk 3 above)
Comment (oleksandr-vovk) [reply backend/Tasque.BLL/Services/FileUploadService.cs]: fixed

File: frontend/src/app/user/components/user-profile/user-profile.component.ts (hunk 6 above)
Comment (oleksandr-vovk) [reply frontend/src/app/user/components/user-profile/user-profile.component.ts]: fixed

File: backend/Tasque.DAL/Migrations/20220817073025_AddedUserAvatar.cs (hunk 5 above)
Comment (oleksandr-vovk) [reply backend/Tasque.DAL/Migrations/20220817073025_AddedUserAvatar.cs:25]: updated

**Review by Anatolii-Ivanov — APPROVED** [review]

**Review by ismv — APPROVED** [review]

**Pull request merged**